src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -2322,67 +2328,259 @@ internal override Binder GetBinder(SyntaxNode node)
 
             public override BoundStatement BindStatement(StatementSyntax node, DiagnosticBag diagnostics)
             {
+                BoundNode boundNode;
+
                 // Check the bound node cache to see if the statement was already bound.
-                BoundStatement synthesizedStatement = _semanticModel.GuardedGetSynthesizedStatementFromMap(node);
+                if (node.SyntaxTree == _semanticModel.SyntaxTree)
+                {
+                    BoundStatement synthesizedStatement = _semanticModel.GuardedGetSynthesizedStatementFromMap(node);
+
+                    if (synthesizedStatement != null)
+                    {
+                        return synthesizedStatement;
+                    }
+
+                    boundNode = TryGetBoundNodeFromMap(node);
```

Comment:
can we declare+initialize `boundNode` here?

Reply:
> can we declare+initialize boundNode here?

Already caught by Chuck, see the comment above.

---
In reply to: [384776472](https://github.com/dotnet/roslyn/pull/41926#discussion_r384776472) [](ancestors = 384776472)